include/boost/url/decode_view.hpp

```diff
@@ -341,6 +341,140 @@ class decode_view
     reference
     back() const noexcept;
 
+    /** Checks if the string begins with the given prefix
+
+        @par Example
+        @code
+        assert( decode_view( "Program%20Files" ).starts_with("Program") );
+        @endcode
+
+        @par Complexity
+        Linear.
+
+        @par Exception Safety
+        Throws nothing.
+    */
+    BOOST_URL_DECL
+    bool
+    starts_with( string_view sv ) const noexcept;
+
+    /** Checks if the string ends with the given prefix
+
+        @par Example
+        @code
+        assert( decode_view( "Program%20Files" ).ends_with("Files") );
+        @endcode
+
+        @par Complexity
+        Linear.
+
+        @par Exception Safety
+        Throws nothing.
+    */
+    BOOST_URL_DECL
+    bool
+    ends_with( string_view sv ) const noexcept;
```

Comment:
I'm not excited about adding more stuff to `decode_view`

Reply:
I guess we could write some free functions. Matching decoded segments without these functions is nearly impossible.

Reply:
we leave it this way for now I have to look closer at it. maybe there's nothing wrong with it.